> You can't use `@QuarkusTest` there.
> 
> You can only use `@QuarkusTest` in our `integration-tests`.
> 
> The alternative is to use `QuarkusUnitTest`

I tried `QuarkusUnitTest` before that, but I got this exception:

```
@RegisterExtension
static final QuarkusUnitTest config = new QuarkusUnitTest().withApplicationRoot((jar) -> jar.addClass(UserResource.class));
```

```
[ERROR] Tests run: 1, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 2.977 s <<< FAILURE! -- in io.quarkus.test.junit.http.TestHTTPResourcePathParamTest
[ERROR] io.quarkus.test.junit.http.TestHTTPResourcePathParamTest -- Time elapsed: 2.977 s <<< ERROR!
java.lang.RuntimeException: org.junit.jupiter.api.extension.TestInstantiationException: Failed to create test instance
	at io.quarkus.test.QuarkusUnitTest.beforeAll(QuarkusUnitTest.java:704)
	at org.junit.jupiter.engine.descriptor.ClassBasedTestDescriptor.lambda$invokeBeforeAllCallbacks$12(ClassBasedTestDescriptor.java:395)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.jupiter.engine.descriptor.ClassBasedTestDescriptor.invokeBeforeAllCallbacks(ClassBasedTestDescriptor.java:395)
	at org.junit.jupiter.engine.descriptor.ClassBasedTestDescriptor.before(ClassBasedTestDescriptor.java:211)
	at org.junit.jupiter.engine.descriptor.ClassBasedTestDescriptor.before(ClassBasedTestDescriptor.java:84)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$6(NodeTestTask.java:148)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$8(NodeTestTask.java:141)
	at org.junit.platform.engine.support.hierarchical.Node.around(Node.java:137)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$9(NodeTestTask.java:139)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.executeRecursively(NodeTestTask.java:138)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:95)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)
	at org.junit.platform.engine.support.hierarchical.SameThreadHierarchicalTestExecutorService.invokeAll(SameThreadHierarchicalTestExecutorService.java:41)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$6(NodeTestTask.java:155)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$8(NodeTestTask.java:141)
	at org.junit.platform.engine.support.hierarchical.Node.around(Node.java:137)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$9(NodeTestTask.java:139)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.executeRecursively(NodeTestTask.java:138)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:95)
	at org.junit.platform.engine.support.hierarchical.SameThreadHierarchicalTestExecutorService.submit(SameThreadHierarchicalTestExecutorService.java:35)
	at org.junit.platform.engine.support.hierarchical.HierarchicalTestExecutor.execute(HierarchicalTestExecutor.java:57)
	at org.junit.platform.engine.support.hierarchical.HierarchicalTestEngine.execute(HierarchicalTestEngine.java:54)
	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.execute(EngineExecutionOrchestrator.java:147)
	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.execute(EngineExecutionOrchestrator.java:127)
	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.execute(EngineExecutionOrchestrator.java:90)
	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.lambda$execute$0(EngineExecutionOrchestrator.java:55)
	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.withInterceptedStreams(EngineExecutionOrchestrator.java:102)
	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.execute(EngineExecutionOrchestrator.java:54)
	at org.junit.platform.launcher.core.DefaultLauncher.execute(DefaultLauncher.java:114)
	at org.junit.platform.launcher.core.DefaultLauncher.execute(DefaultLauncher.java:86)
	at org.junit.platform.launcher.core.DefaultLauncherSession$DelegatingLauncher.execute(DefaultLauncherSession.java:86)
	at org.apache.maven.surefire.junitplatform.LazyLauncher.execute(LazyLauncher.java:56)
	at org.apache.maven.surefire.junitplatform.JUnitPlatformProvider.execute(JUnitPlatformProvider.java:184)
	at org.apache.maven.surefire.junitplatform.JUnitPlatformProvider.invokeAllTests(JUnitPlatformProvider.java:148)
	at org.apache.maven.surefire.junitplatform.JUnitPlatformProvider.invoke(JUnitPlatformProvider.java:122)
	at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:385)
	at org.apache.maven.surefire.booter.ForkedBooter.execute(ForkedBooter.java:162)
	at org.apache.maven.surefire.booter.ForkedBooter.run(ForkedBooter.java:507)
	at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:495)
	Suppressed: java.lang.NoClassDefFoundError: io/quarkus/test/common/GroovyCacheCleaner
		at io.quarkus.test.QuarkusUnitTest.afterAll(QuarkusUnitTest.java:762)
		at org.junit.jupiter.engine.descriptor.ClassBasedTestDescriptor.lambda$invokeAfterAllCallbacks$18(ClassBasedTestDescriptor.java:461)
		at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
		at org.junit.jupiter.engine.descriptor.ClassBasedTestDescriptor.lambda$invokeAfterAllCallbacks$19(ClassBasedTestDescriptor.java:461)
		at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)
		at org.junit.jupiter.engine.descriptor.ClassBasedTestDescriptor.invokeAfterAllCallbacks(ClassBasedTestDescriptor.java:461)
		at org.junit.jupiter.engine.descriptor.ClassBasedTestDescriptor.after(ClassBasedTestDescriptor.java:235)
		at org.junit.jupiter.engine.descriptor.ClassBasedTestDescriptor.after(ClassBasedTestDescriptor.java:84)
		at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$7(NodeTestTask.java:161)
		at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
		at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$8(NodeTestTask.java:161)
		... 35 more
	Caused by: java.lang.ClassNotFoundException: io.quarkus.test.common.GroovyCacheCleaner
		at java.base/jdk.internal.loader.BuiltinClassLoader.loadClass(BuiltinClassLoader.java:641)
		at java.base/jdk.internal.loader.ClassLoaders$AppClassLoader.loadClass(ClassLoaders.java:188)
		at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:521)
		... 46 more
Caused by: org.junit.jupiter.api.extension.TestInstantiationException: Failed to create test instance
	at io.quarkus.test.QuarkusUnitTest.beforeAll(QuarkusUnitTest.java:676)
	... 43 more
Caused by: java.lang.RuntimeException: java.lang.reflect.InvocationTargetException
	at io.quarkus.runner.bootstrap.RunningQuarkusApplicationImpl.instance(RunningQuarkusApplicationImpl.java:90)
	at io.quarkus.test.QuarkusUnitTest.beforeAll(QuarkusUnitTest.java:668)
	... 43 more
Caused by: java.lang.reflect.InvocationTargetException
	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:119)
	at java.base/java.lang.reflect.Method.invoke(Method.java:578)
	at io.quarkus.runner.bootstrap.RunningQuarkusApplicationImpl.instance(RunningQuarkusApplicationImpl.java:85)
	... 44 more
Caused by: java.lang.IllegalStateException: Unable to locate CDIProvider
	at jakarta.enterprise.inject.spi.CDI.findAllProviders(CDI.java:139)
	at jakarta.enterprise.inject.spi.CDI.getCDIProvider(CDI.java:92)
	at jakarta.enterprise.inject.spi.CDI.current(CDI.java:65)
	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:104)
	... 46 more
```

--- a/extensions/resteasy-reactive/quarkus-resteasy-reactive/deployment/src/test/java/io/quarkus/resteasy/reactive/server/test/StringTestHTTPResourceWithPathParamsTest.java
+++ b/extensions/resteasy-reactive/quarkus-resteasy-reactive/deployment/src/test/java/io/quarkus/resteasy/reactive/server/test/StringTestHTTPResourceWithPathParamsTest.java
@@ -1,0 +1,45 @@
+package io.quarkus.resteasy.reactive.server.test;
+
+import static io.restassured.RestAssured.given;
+import static org.hamcrest.Matchers.equalTo;
+
+import jakarta.ws.rs.GET;
+import jakarta.ws.rs.Path;
+import jakarta.ws.rs.PathParam;
+import jakarta.ws.rs.Produces;
+import jakarta.ws.rs.core.MediaType;
+
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.RegisterExtension;
+
+import io.quarkus.test.QuarkusUnitTest;
+import io.quarkus.test.common.http.TestHTTPEndpoint;
+import io.quarkus.test.common.http.TestHTTPResource;
+
+public class StringTestHTTPResourceWithPathParamsTest {
+
+    @RegisterExtension
+    static final QuarkusUnitTest test = new QuarkusUnitTest().withApplicationRoot((jar) -> jar.addClass(UserResource.class));
+
+    @TestHTTPEndpoint(UserResource.class)
+    @TestHTTPResource("{userId}/order/{orderId}")
+    String getUserOrderUrl;
+
+    @Test
+    void testGettingUserOrder() {
+        int userId = 123;
+        int orderId = 456;
+        given().when().get(getUserOrderUrl, userId, orderId)
+                .then().statusCode(200).body(equalTo(String.format("Order (%d) of user (%d)", userId, orderId)));
+    }
+
+    @Path("/user")
+    public static class UserResource {
+        @Path("{userId}/order/{orderId}")
+        @GET
+        @Produces(MediaType.TEXT_PLAIN)
+        public String getUserOrder(@PathParam("userId") int userId, @PathParam("orderId") int orderId) {
+            return String.format("Order (%d) of user (%d)", userId, orderId);
+        }
+    }
+}

--- a/test-framework/common/src/main/java/io/quarkus/test/common/http/StringTestHTTPResourceProvider.java
+++ b/test-framework/common/src/main/java/io/quarkus/test/common/http/StringTestHTTPResourceProvider.java
@@ -1,7 +1,6 @@
 package io.quarkus.test.common.http;
 
 import java.lang.reflect.Field;
-import java.net.URI;
 
 public class StringTestHTTPResourceProvider implements TestHTTPResourceProvider<String> {
     @Override
@@ -10,7 +9,7 @@ public class StringTestHTTPResourceProvider implements TestHTTPResourceProvider<
     }
 
     @Override
-    public String provide(URI testUri, Field field) {
-        return testUri.toASCIIString();
+    public String provide(String testUri, Field field) {
+        return testUri;
     }
 }

--- a/test-framework/common/src/main/java/io/quarkus/test/common/http/TestHTTPResourceManager.java
+++ b/test-framework/common/src/main/java/io/quarkus/test/common/http/TestHTTPResourceManager.java
@@ -1,7 +1,6 @@
 package io.quarkus.test.common.http;
 
 import java.lang.reflect.Field;
-import java.net.URI;
 import java.util.Collections;
 import java.util.HashMap;
 import java.util.List;
@@ -102,7 +101,7 @@ public class TestHTTPResourceManager {
                     }
                     f.setAccessible(true);
                     try {
-                        f.set(testCase, provider.provide(new URI(val), f));
+                        f.set(testCase, provider.provide(val, f));
                     } catch (Exception e) {
                         throw new RuntimeException(e);
                     }

--- a/test-framework/common/src/main/java/io/quarkus/test/common/http/TestHTTPResourceProvider.java
+++ b/test-framework/common/src/main/java/io/quarkus/test/common/http/TestHTTPResourceProvider.java
@@ -1,7 +1,6 @@
 package io.quarkus.test.common.http;
 
 import java.lang.reflect.Field;
-import java.net.URI;
 
 public interface TestHTTPResourceProvider<T> {
 
@@ -13,6 +12,6 @@ public interface TestHTTPResourceProvider<T> {
      * Note that there is no need to directly call set() on the field, it is only provided
      * to allow you to examine the generic type and any additional annotations.
      */
-    T provide(URI testUri, Field field);
+    T provide(String testUri, Field field);
 
 }

--- a/test-framework/common/src/main/java/io/quarkus/test/common/http/URITestHTTPResourceProvider.java
+++ b/test-framework/common/src/main/java/io/quarkus/test/common/http/URITestHTTPResourceProvider.java
@@ -2,6 +2,7 @@ package io.quarkus.test.common.http;
 
 import java.lang.reflect.Field;
 import java.net.URI;
+import java.net.URISyntaxException;
 
 public class URITestHTTPResourceProvider implements TestHTTPResourceProvider<URI> {
     @Override
@@ -10,7 +11,11 @@ public class URITestHTTPResourceProvider implements TestHTTPResourceProvider<URI
     }
 
     @Override
-    public URI provide(URI testUri, Field field) {
-        return testUri;
+    public URI provide(String testUri, Field field) {
+        try {
+            return new URI(testUri);
+        } catch (URISyntaxException e) {
+            throw new RuntimeException(e);
+        }
     }
 }

--- a/test-framework/common/src/main/java/io/quarkus/test/common/http/URLTestHTTPResourceProvider.java
+++ b/test-framework/common/src/main/java/io/quarkus/test/common/http/URLTestHTTPResourceProvider.java
@@ -3,6 +3,7 @@ package io.quarkus.test.common.http;
 import java.lang.reflect.Field;
 import java.net.MalformedURLException;
 import java.net.URI;
+import java.net.URISyntaxException;
 import java.net.URL;
 
 public class URLTestHTTPResourceProvider implements TestHTTPResourceProvider<URL> {
@@ -12,10 +13,10 @@ public class URLTestHTTPResourceProvider implements TestHTTPResourceProvider<URL
     }
 
     @Override
-    public URL provide(URI testUri, Field field) {
+    public URL provide(String testUri, Field field) {
         try {
-            return testUri.toURL();
-        } catch (MalformedURLException e) {
+            return new URI(testUri).toURL();
+        } catch (MalformedURLException | URISyntaxException e) {
             throw new RuntimeException(e);
         }
     }

